### PR TITLE
fix: always set feature flag

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:
https://d2iq.atlassian.net/browse/D2IQ-93868

upstream seems to always sets these flags https://github.com/kubernetes-sigs/image-builder/blob/02df45969409c7f18f2cf7e63b70c4dc0e2fa442/images/capi/ansible/roles/providers/tasks/main.yml#L105

i started vm with ubuntu 20.04 and ran cloud-init version 

```bash
root@ip-10-0-122-187:/var/snap/amazon-ssm-agent/6312# cloud-init --version
/usr/bin/cloud-init 22.3.4-0ubuntu1~20.04.1
```

so i also updated the regex and the version comparision command as well.

See this conversation in the kubernetes slack for why we included this version check in the first place https://kubernetes.slack.com/archives/CD6U2V71N/p1629737897163400 

**Which issue(s) does this PR fix?**:

* https://jira.d2iq.com/browse/D2IQ-NUMBER

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: ansible/roles/providers/tasks/misc.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@
     register: system_cloud_init_cmd
   - name: Set cloud-init version fact
     set_fact:
-      system_cloud_init_version: "{{ system_cloud_init_cmd.stdout | regex_replace('^.*cloud-init\\s(?P<version>\\d+?\\.?\\d+)-.*$', '\\g<version>') }}"
+      system_cloud_init_version: "{{ system_cloud_init_cmd.stdout | regex_replace('^.*cloud-init\\s(?P<version>\\d+?\\.?\\d+)(-.*)?$', '\\g<version>') }}"
```
Comment (faiq) [inline ansible/roles/providers/tasks/misc.yml:60]: regex tested here https://regex101.com/r/2Xc5Vx/1

**Review by dkoshkin — APPROVED** [review]

**Comment by faiq** [comment]
tests failing probably because of the regex?

File: ansible/roles/providers/tasks/misc.yml (hunk 1 above)
Comment (alejandroEsc) [reply ansible/roles/providers/tasks/misc.yml]: thank you for posting the regex test, awesome.

**Review by alejandroEsc — APPROVED** [review]
lgtm

**Comment by alejandroEsc** [comment]
rerunning e2e sles 15 build as it failed

**Pull request closed without merging**